samples/snippets/test_sample_templates.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import uuid
+
+import google.auth
+
+from sample_templates import (
+    create_template,
+    create_template_from_instance,
+    create_template_with_subnet,
+    delete_instance_template,
+    list_instance_templates,
+)
+
+from test_sample_start_stop import compute_instance
+# Needed to make the flake8 happy.
+assert compute_instance
+
+PROJECT = google.auth.default()[1]
+
+INSTANCE_ZONE = "europe-central2-b"
+
+
+def test_create_instance():
+    template_name = "i" + uuid.uuid4().hex[:10]
+    template = create_template(PROJECT, template_name)
+
+    try:
+        assert template.name == template_name
+        assert template.properties.disks[0].initialize_params.disk_size_gb == 250
+        assert (
+            "debian-11" in template.properties.disks[0].initialize_params.source_image
+        )
+        assert (
+            template.properties.network_interfaces[0].name == "global/networks/default"
+        )
+        assert template.properties.machine_type == "e2-standard-4"
+    finally:
+        delete_instance_template(PROJECT, template_name)
```

Comment:
Please use a fixture here instead: https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#testing-guidelines

Reply:
I always thought of fixtures as a method of creating all the necessary objects that are not part of the test itself. To create dependencies the test requires. In this case, what is being tested is the creation of a template (and as side effect deletion of it), so I put that inside the test function.

Using a fixture here would basically move the logic of what is being tested outside the scope of the test function. The test itself would test the results of some action, but wouldn't be itself responsible for performing that action.

Reply:
This test is testing the results of `create_template`, not `delete_instance_template`. 

You can create different types of fixtures that only do certain behavior, such here you might want to do something like:

```python
@pytest.fixture
def creatable_instance():
    template_name = "i" + uuid.uuid4().hex[:10]
    yield template_name
    delete_instance_template(PROJECT, template_name)
```

Reply:
OK, I have updated the tests to follow your way of using fixtures.